### PR TITLE
IPv4/single: applied MISRA to FreeRTOS_ARP.c

### PR DESCRIPTION
**Pull request by htibosch**
From time to time we check if the code is still complying to the MISRA rules.

This PR presents some related change for FreeRTOS_ARP.c.

By submitting this pull request, I confirm that you can use, modify, copy, and redistribute this contribution, under the terms of your choice.

### CODE REVIEW

File: FreeRTOS_ARP.c [hunk 1]
```diff
@@ -638,17 +637,19 @@ eARPLookupResult_t eARPGetCacheEntry( uint32_t * pulIPAddress,
...
+    else if( *ipLOCAL_IP_ADDRESS_POINTER == 0U )
     {
         /* The IP address has not yet been assigned, so there is nothing that
          * can be done. */
         eReturn = eCantSendPacket;
     }
     else if( *ipLOCAL_IP_ADDRESS_POINTER == *pulIPAddress )
     {
+        const void * pvCopySource = ( const void * ) ipLOCAL_MAC_ADDRESS;
+        void * pvCopyDest = ( void * ) pxMACAddress->ucBytes;
```
Comment (AniruddhaKanhere) [inline FreeRTOS_ARP.c:649]: Why do we need this if both data types are similar?

Comment (htibosch) [reply FreeRTOS_ARP.c:649]: Possibly not, I will check that.

**Review by AniruddhaKanhere — COMMENTED** [review]
Thank you @htibosch for this PR. I approve of this and we can merge these changes in the big PR for all the sources.

I think we can close this now

**Comment by htibosch** [comment]
The changes proposed in this PR also appear in PR #357, and they will be merged to main from there.
Closing this PR now.

**Pull request closed without merging**